### PR TITLE
Add Governance module foundation

### LINKED ISSUE #8862: Prepare general structure of Governance module
### Description

Prepare a folder structure and files for governance module:
- module
- endpoint
- method
- stores
- events
- internal_methods
- schema
- constants
- index

### Acceptance Criteria

- No specific logic should be added, just template to implement the Governance module should be sufficient.

### PR DESCRIPTION
**Pull request by bobanm**
### What was the problem?

This PR resolves #8862

### How was it solved?

Added stubs for:

- Constants
- Types
- Module
- Method
- Endpoint
- Index
- Proposal Created event
- Proposal Creation Failed event
- Proposal Outcome event
- Proposal Quorum Checked event
- Proposal Voted event
- Config store
- Index store
- Proposals store
- Votes store
- Create Proposal command

### How was it tested?

Not applicable

### CODE REVIEW

File: framework/src/modules/governance/events/proposal_created.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import { BaseEvent } from '../../base_event';
+import { ProposalType } from '../types';
+
+interface ProposalCreatedEventData {
+	creator: Buffer;
+	index: number;
+	type: ProposalType;
+}
+
+export class ProposalCreatedEvent extends BaseEvent<ProposalCreatedEventData> {}
```
Comment (shuse2) [inline framework/src/modules/governance/events/proposal_created.ts:24]: schema is missing?

File: framework/src/modules/governance/stores/index.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+import { BaseStore } from '../../base_store';
+
+export interface IndexStoreData {
```
Comment (shuse2) [inline framework/src/modules/governance/stores/index.ts:16]: index has special meaning so maybe we can change the name to `ProposalIndex`?

File: framework/src/modules/governance/types.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import { ImmutableMethodContext, MethodContext } from '../../state_machine';
+
+export enum ProposalType {
```
Comment (shuse2) [inline framework/src/modules/governance/types.ts:17]: `const enum` would be better for efficiency?

File: framework/src/modules/governance/events/proposal_created.ts (hunk 1 above)
Comment (bobanm) [reply framework/src/modules/governance/events/proposal_created.ts:24]: I was planning to add the event schema and event logging as part of event quest #8869

File: framework/src/modules/governance/stores/index.ts (hunk 2 above)
Comment (bobanm) [reply framework/src/modules/governance/stores/index.ts]: True. I also didn't like calling it `index`.

**Review by Incede — APPROVED** [review]

**Review by shuse2 — APPROVED** [review]

**Pull request merged**